### ISSUE TITLE
Error when accessing threads of a channel that one has not joined

### ISSUE THREAD

**Issue by Prateek93a** [RocketChat/Rocket.Chat.ReactNative#1444]
- Your Rocket.Chat Experimental app version: 1.25.0.9999...
- Your Rocket.Chat server version: 3.0.0-develop open.rocket.chat
- Device (or Simulator) you're running with: Android P Sumsung Galaxy J6

![ezgif com-video-to-gif(6)](https://user-images.githubusercontent.com/44807945/70388421-92379280-19d7-11ea-8e5e-f1ee40020a6f.gif)

As the gif shows, the app shows shows error, when one try to access threads of a channel that he has not joined.

**Comment by Prateek93a** [comment]
@diegolmello 
app > views > ThreadMessagesView> index.js > line no 100 inside subsribeData() function. when I comment out the code on line 100, the error disappears, and everything works fine, but I am not able to understand why this is so. 


![Screenshot from 2019-12-26 12-21-50](https://user-images.githubusercontent.com/44807945/71462947-ecf42b00-27da-11ea-9276-4eb2034a0e0c.png)

**Comment by djorkaeffalexandre** [comment]
@Prateek93a try change `findAndObserve` to two functions like:
```
try {
  const db = database.active;
  const subCollection = await db.collections.get('subscriptions');
  const subscription = await subCollection.find(this.rid);
  const observable = subscription.observe();
  this.subSubscription = observable
    .subscribe(changes => console.log(changes));
catch (e) {
  console.log('subscription not found');
}
```
Unfortunately the function `findAndObserve` not works fine..

**Comment by Prateek93a** [comment]
@djorkaeffalexandre Yes Sir your solution worked, since the `findAndObserve` was not throwing error, the app was crashing. Now after splitting the function, the error got thrown and logged out in the console, without crashing the app. So the solution is working.

**Comment by djorkaeffalexandre** [comment]
;)

**Comment by djorkaeffalexandre** [comment]
hey @Prateek93a, can you confirm if it's happening yet?
I followed your gif [here](https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/1494#issuecomment-587361708) and can't reproduce any error.

**Comment by Prateek93a** [comment]
@djorkaeffalexandre I can't reproduce it. I think it was fixed by #1997. Closing it.
